### DISCUSSION TITLE
Massive RAM spike on loading certain maps

### DISCUSSION THREAD

**Discussion by Morilli** [ppy/osu#15854] [Q&A]
Some maps, when imported, selected in song select or actually played, will cause a huge rise in RAM usage (I suspect it's probably difficulty calculation or something that's trying to emulate the map). Example beatmap that should show this behavior: https://osu.ppy.sh/beatmapsets/592138#osu/1277064
Just importing this map from within lazer, I am seeing a ram (allocated memory) increase from ~1.1 GB to ~6GB. Of course this drops again after waiting for garbage collection or manually initiating it, but this still feels wrong.
CTRL+F2 GC statictics while having the top diff selected (and after a big of GC already happening):
![image](https://user-images.githubusercontent.com/35152647/143825457-9aaad2a1-0ba3-4e81-a617-ec119372d4da.png)

osu!lazer version used: 2021.1127.0-lazer (desktop, windows 11 22494.1000)

**Comment by peppy** [comment]
This probably falls under the existing issue for aspire/gamebreaking beatmaps.
